omod-1.9/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_9/ObsResource1_9.java

```diff
@@ -31,7 +33,8 @@
 	 */
 	@PropertySetter("concept")
 	public static void setConcept(Obs obs, Object value) {
-		obs.setConcept(new ConceptResource1_9().getByUniqueId((String) value));
+		String uuid = value instanceof Map ? ((Map) value).get("uuid").toString() : (String) value;
+		obs.setConcept(new ConceptResource1_9().getByUniqueId(uuid));
```

Comment:
Same here, you should be delegating to the resource

Reply:
Actually, this is close to what I was suggesting, you only need to change it to look up the resource rather than instantiating one from here

Reply:
If you think about, what the code in this method is doing is already taken care of by the framework and now all is doing is causing bugs like the one you are running into, the fix is to remove it anyways.